Add code tabs for _tour/compound-types (#2552)
scala/docs.scala-lang: _tour/compound-types.md

{% tabs compound-types_2 class=tabs-scala-version %}
{% tab 'Scala 2' for=compound-types_2 %}
```scala mdoc:fail
def cloneAndReset(obj: ?): Cloneable = {
  val cloned = obj.clone()
  obj.reset
  cloned
}
```
{% endtab %}
{% tab 'Scala 3' for=compound-types_2 %}
```scala
def cloneAndReset(obj: ?): Cloneable =
  val cloned = obj.clone()
  obj.reset
  cloned
```
{% endtab %}
{% endtabs %}

The question arises what the type of the parameter `obj` is. If it's `Cloneable` then the object can be `clone`d, but not `reset`; if it's `Resetable` we can `reset` it, but there is no `clone` operation. To avoid type casts in such a situation, we can specify the type of `obj` to be both `Cloneable` and `Resetable`.
{% tabs compound-types_3 class=tabs-scala-version %}
{% tab 'Scala 2' for=compound-types_3 %}
This compound type is written like this in Scala: `Cloneable with Resetable`.

Here's the updated function:
```scala mdoc:fail
def cloneAndReset(obj: Cloneable with Resetable): Cloneable = {
  //...
}
```
Compound types can consist of several object types and they may have a single refinement which can be used to narrow the signature of existing object members.

> The general form is: `A with B with C ... { refinement }`

Reviewer: I suggest we update like this:
```suggestion
Note that you can have more than two types: `A with B with C with ...`.
This means the same as thing as `(...(A with B) with C`) with ... )`
```

{% endtab %}

> {% tab 'Scala 3' for=compound-types_2 %}

Reviewer 2: ```suggestion
{% tab 'Scala 3' for=compound-types_3 %}
```

This compound type is written like this in Scala: `Cloneable & Resetable`.

Here's the updated function:
```scala
def cloneAndReset(obj: Cloneable & Resetable): Cloneable = {
  //...
}
```
Compound types can consist of several object types and they may have a single refinement which can be used to narrow the signature of existing object members.

> The general form is: `A & B & C ... { refinement }`

Reviewer: I suggest we update like this:
```suggestion
Note that you can have more than two types: `A & B & C & ...`. 
And `&` is associative, so parentheses can be added around any part without changing the meaning.
```
